src/mlpack/methods/sparse_coding/sparse_coding_impl.hpp

```diff
@@ -106,6 +106,7 @@ void SparseCoding::Train(
   }
 
   Timer::Stop("sparse_coding");
+  return lastObjVal;
```

Comment:
Should this be `curObjVal`?

Reply:
No actually, curObjVal is only in the scope of for loop , so we cannot return that. But lastObjVal is in the function scope and in the for loop we can see that we are copying curObjVal in lastObjVal (line 105). So I thought instead of declaring curObjVal outside for loop. It's better to return lastObjVal as it's getting updated with curObjVal at the end of for loop.

Reply:
Ok, but in this case take a look at line 101---if it converges, then `lastObjVal` is not up to date.  So I guess we need to update it there.